vpr/src/route/lookahead_profiler.cpp

```diff
@@ -0,0 +1,169 @@
+
+        // Get the difference in the coordinates in the current and sink nodes.
+        // Note: we are not using util::get_xy_deltas() because this always gives positive values
+        // unless the current node is the source node. Using util::get_adjusted_rr_position therefore
+        // gives us more information.
+        auto [from_x, from_y] = util::get_adjusted_rr_position(curr_inode);
+        auto [to_x, to_y] = util::get_adjusted_rr_position(sink_inode);
+
+        int delta_x = to_x - from_x;
+        int delta_y = to_y - from_y;
+
+        // Get the current node's type and length
+        std::string node_type_str = rr_graph.node_type_string(curr_inode);
+        std::string node_length = (node_type_str == "CHANX" || node_type_str == "CHANY")
+                                      ? std::to_string(rr_graph.node_length(curr_inode))
+                                      : "--";
+
+        /* Write out all info */
+
+        lookahead_verifier_csv_ << iteration << ",";            // iteration no.
+        lookahead_verifier_csv_ << source_inode << ",";         // source node
+        lookahead_verifier_csv_ << sink_inode << ",";           // sink node
+        lookahead_verifier_csv_ << block_name << ",";           // sink block name
+        lookahead_verifier_csv_ << atom_block_model << ",";     // sink atom block model
+        lookahead_verifier_csv_ << cluster_block_type << ",";   // sink cluster block type
+        lookahead_verifier_csv_ << tile_height << ",";          // sink cluster tile height
+        lookahead_verifier_csv_ << tile_width << ",";           // sink cluster tile width
+        lookahead_verifier_csv_ << curr_inode << ",";           // current node
+        lookahead_verifier_csv_ << node_type_str << ",";        // node type
+        lookahead_verifier_csv_ << node_length << ",";          // node length
+        lookahead_verifier_csv_ << nodes_from_sink << ",";      // num. nodes from sink
+        lookahead_verifier_csv_ << delta_x << ",";              // delta x
+        lookahead_verifier_csv_ << delta_y << ",";              // delta y
+        lookahead_verifier_csv_ << actual_cost << ",";          // actual cost
+        lookahead_verifier_csv_ << actual_delay << ",";         // actual delay
+        lookahead_verifier_csv_ << actual_congestion << ",";    // actual congestion
+        lookahead_verifier_csv_ << lookahead_cost << ",";       // predicted cost
+        lookahead_verifier_csv_ << lookahead_delay << ",";      // predicted delay
+        lookahead_verifier_csv_ << lookahead_congestion << ","; // predicted congestion
+        lookahead_verifier_csv_ << cost_params.criticality;     // criticality
+        lookahead_verifier_csv_ << "\n";
+    }
+}
+
+void LookaheadProfiler::clear() {
+    net_pin_blocks_.clear();
+    sink_atom_block_.clear();
+    sink_cluster_block_.clear();
+    tile_types_.clear();
```

Comment:
I guess clear() does not deallocate memory. Vtr has a utility function that swaps a given container with an empty one to force memory deallocation. If I remember correctly, it is called release_memory()